packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -80,14 +74,14 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 				),
 				hasBlockStyles: !! styles?.length,
 				icon: _icon,
-				blockTitle:
-					reusableBlockTitle || getBlockType( firstBlockName ).title,
+				blockTitle: getBlockType( firstBlockName ).title,
 			};
 		},
 		[ clientIds, blocks, blockInformation?.icon ]
 	);
 
 	const isReusable = blocks.length === 1 && isReusableBlock( blocks[ 0 ] );
+	const isTemplate = blocks.length === 1 && isTemplatePart( blocks[ 0 ] );
```

Comment:
Leaving it a bit more explicit for now because down the line we might want to abstract this into something a block author can opt into (`showTitleInToolbar`), but I'd like to see how it fares in the meantime.